pyuvm/s19_pyvum_reg_field.py

```diff
@@ -1,212 +1,208 @@
-        ## i.e. not "RW", "WRC", "WRS", "WO", "W1", "WO1"
-        if enable_pyvsc == True:
-            if self._access in ["RW", "WRC", "WRS", "WO", "W1", "WO1"]:
-                self._desidered.rand_mode = True
-            else:
-                self._desidered.rand_mode = False
-        else:
-            self._desidered = 0
-        ## Check if size is 0
-        if(self._size == 0):
-            raise UVMFatalError(f"{self._header} Size of a filed cannot be 0 MINIMUN allowed is 1")
-        ## Check if policy is a valid policy
-        if(self._access not in self.access_list):
+        self._error_on_write = False
+
+    # configure
+    def configure(self, parent: uvm_reg, size: int, lsb_pos: int, access: str, is_volatile: bool, reset: int):
+        self._parent = parent
+        self._size = size
+        self._lsb_pos = lsb_pos
+        self._access = access
+        self._is_volatile = is_volatile
+        self._reset = reset
+        self._config_done = False
+        self._compare = check_t
+        # Additional Checking
+        # Ignore randomization if the field is known not to be writeable
+        # i.e. not "RW", "WRC", "WRS", "WO", "W1", "WO1"
+        # TODO: add eventually support for PYVSC for randomization
+        self._desidered = 0
+        # Check if size is 0
+        if (self._size == 0):
+            uvm_fatal(self._header, "Size of a filed cannot be 0 MINIMUN allowed is 1")
+        # Check if policy is a valid policy
+        if (self._access not in self.access_list):
             self._access = "NOACCESS"
-        ## These 2 flags cannot change for fields since are part of the parent register
+        # These 2 flags cannot
+        # change for fields since are part of the parent register
         self._error_on_read = self._parent.throw_error_on_read
-        self._error_on_write= self._parent.throw_error_on_write         
-        ## one configure is called let's unlock
-        self.field_unlock()
-        ## check the reset value is not beyond the MAX
-        if(self._reset >= (2**self._size)):
-            error_out(self._header,f"Reset value for REG feild : {self._name} is [{self._reset}] is beyond the MAX valoue given by the size [{((2**self._size)-1)}]")
-        ## Add field
-        parent._add_field(self)   
-
-    ## lock method meaning the configursation is done and we can unlock all the internal methods
+        self._error_on_write = self._parent.throw_error_on_write  
+        # one configure is called let's lock
+        self.field_lock()
+        # check the reset value is not beyond the MAX
+        if (self._reset >= (2**self._size)):
+            uvm_error(self._header, f"Reset value for REG feild : {self._name} is [{self._reset}] is beyond the MAX valoue given by the size [{((2**self._size)-1)}]")
+        # Add field
+        parent._add_field(self)
+
+    # lock method meaning the configursation is
+    # done and we can unlock all the internal methods
```

Comment:
in order to get access to internal methods anytime a new field is added we do lock it since the the configure should be called after a register is declared inside a reg class. Added checks on size and reset value never beyond the max value (reset value and all the internal properties are `int` hence should be constrained -- this will fixed by using the `BinaryValue` instead